api-authorizations/src/main/scala/com/normation/plugins/apiauthorizations/UserTokenApiDefinition.scala

```diff
@@ -128,67 +111,172 @@ class UserApi(
-          RestUtils.toJsonError(None, e.messageChain)(schema.name, true)
-      }
+      writeApi
+        .save(account, ModificationId(uuidGen.newUuid), authzToken.actor)
+        .map(RestAccountsResponse.fromUnredacted(_, secret))
+        .chainError(s"Error when trying to save user '${authzToken.actor.name}' API token")
+        .toLiftResponseOne(params, schema, None)
     }
   }
 
   object DeleteApiToken extends LiftApiModule0 {
-    val schema        = API.DeleteApiToken
-    val restExtractor = api.restExtractor
+    val schema = API.DeleteApiToken
     def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
-      writeApi.delete(ApiAccountId(authzToken.actor.name), ModificationId(uuidGen.newUuid), authzToken.actor).toBox match {
-        case Full(account) =>
-          val accounts: JValue = ("accounts" -> ("id" -> account.value))
-          RestUtils.toJsonResponse(None, accounts)(schema.name, true)
-
-        case eb: EmptyBox =>
-          val e = eb ?~! s"Error when trying to delete user '${authzToken.actor.name}' API token"
-          RestUtils.toJsonError(None, e.messageChain)(schema.name, true)
-      }
+      writeApi
+        .delete(ApiAccountId(authzToken.actor.name), ModificationId(uuidGen.newUuid), authzToken.actor)
+        .map(RestAccountIdResponse(_))
+        .chainError(s"Error when trying to delete user '${authzToken.actor.name}' API token")
+        .toLiftResponseOne(params, schema, None)
     }
   }
 
   object UpdateApiToken extends LiftApiModule0 {
-    val schema        = API.UpdateApiToken
-    val restExtractor = api.restExtractor
+    val schema = API.UpdateApiToken
     def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
-      readApi.getById(ApiAccountId(authzToken.actor.name)).toBox match {
-        case Full(Some(account)) =>
-          val filtered = account.copy(token = if (account.token.isHashed) {
-            // Don't send hashes
-            ApiToken("")
-          } else {
-            account.token
-          })
-          val accounts: JValue = ("accounts" -> JArray(List(filtered.toJson)))
-          RestUtils.toJsonResponse(None, accounts)(schema.name, true)
+      readApi
+        .getById(ApiAccountId(authzToken.actor.name))
+        .map(RestAccountsResponse.fromRedacted(_))
+        .chainError(s"Error when trying to get user '${authzToken.actor.name}' API token")
+        .toLiftResponseOne(params, schema, None)
+    }
+  }
 
-        case Full(None) =>
-          val accounts: JValue = ("accounts" -> JArray(Nil))
-          RestUtils.toJsonResponse(None, accounts)(schema.name, true)
+}
+
+object UserApi {
+
+  final case class RestApiAccount(
+      id:                              ApiAccountId,
+      name:                            ApiAccountName,
+      token:                           ApiToken,
```

Comment:
it would make sense to not use `ApiToken`, but a new `ClearTextToken(value: String)` or something, so that it's not the same structure send, and we don't forget to zero it during a refactoring.